test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -1754,89 +1929,79 @@ public void updatingBlankExternalIdFails() {
-        verify(externalIdService).assignExternalId(STUDY, null, HEALTH_CODE);
+        verify(externalIdService, never()).commitAssignExternalId(any());
+        assertEquals(EXTERNAL_ID, account.getExternalId());
     }
-    
+
     @Test
     public void sameManagedExternalIdOnUpdateIgnored() {
         mockHealthCodeAndAccountRetrieval();
         STUDY.setExternalIdValidationEnabled(true);
         ExternalIdentifier identifier = ExternalIdentifier.create(STUDY.getStudyIdentifier(), EXTERNAL_ID);
-        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID)).thenReturn(identifier);
+        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID, false)).thenReturn(identifier);
         
         participantService.updateParticipant(STUDY, PARTICIPANT);
         
         assertEquals(EXTERNAL_ID, account.getExternalId());
-        verify(externalIdService, never()).assignExternalId(STUDY, EXTERNAL_ID, HEALTH_CODE);
+        verify(externalIdService, never()).commitAssignExternalId(any());
     }
-    
-    @Test
-    public void removingManagedExternalIdWorks() {
-        mockHealthCodeAndAccountRetrieval();
-        STUDY.setExternalIdValidationEnabled(true);
-        ExternalIdentifier identifier = ExternalIdentifier.create(STUDY.getStudyIdentifier(), EXTERNAL_ID);
-        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), "newExternalId")).thenReturn(identifier);
-        
-        // This record has a different external ID than the mocked accound
-        StudyParticipant participant = new StudyParticipant.Builder().copyOf(PARTICIPANT)
-                .withExternalId(null).build();
-        participantService.updateParticipant(STUDY, participant);
-        
-        assertEquals(null, account.getExternalId());
-        verify(externalIdService).unassignExternalId(STUDY, EXTERNAL_ID, HEALTH_CODE);
-        verify(externalIdService).assignExternalId(STUDY, null, HEALTH_CODE);
-    }
-    
+
+    // Removed because you can no longer simply remove an external ID
+    // public void removingManagedExternalIdWorks();
+
     @Test
     public void createUnmanagedExternalIdWillAssign() {
         mockHealthCodeAndAccountRetrieval();
         STUDY.setExternalIdValidationEnabled(false);
-        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID)).thenReturn(null);
+        
+        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID, false)).thenReturn(null);
         
         participantService.createParticipant(STUDY, PARTICIPANT, false);
         
-        verify(externalIdService).assignExternalId(STUDY, EXTERNAL_ID, HEALTH_CODE);
+        verify(accountDao).createAccount(any(), accountCaptor.capture(), any());
+        assertEquals(EXTERNAL_ID, accountCaptor.getValue().getExternalId());
+        
+        verify(externalIdService).commitAssignExternalId(null);
```

Comment:
This seems like a change in behavior. Is this unintentional? Or do we no longer support unmanaged external IDs?

Reply:
The test verifies that if management is turned off and you thus do not retrieve an external ID object, the method correctly assigns your unmanaged ID to the externalId field but does not do an update of the external ID table (it just passes null back which does nothing). BUT this functionality *is* going away after migration, once we rely entirely on the accountSubstudies association. No one has used unmanaged external IDs that I am aware of.